src/components/CheckboxItem.tsx

```diff
@@ -0,0 +1,24 @@
+import clsx from "clsx";
+import { checkIcon } from "./icons";
+
+import "./CheckboxItem.scss";
+
+export const CheckboxItem: React.FC<{
+  checked: boolean;
+  onChange: (checked: boolean) => void;
+}> = ({ children, checked, onChange }) => {
+  return (
+    <div
+      className={clsx("Checkbox", { "is-checked": checked })}
+      onClick={(event) => {
+        onChange(!checked);
+        ((event.currentTarget as HTMLDivElement).querySelector(
+          ".Checkbox-box",
+        ) as HTMLButtonElement).focus();
+      }}
+    >
+      <button className="Checkbox-box">{checkIcon}</button>
```

Comment:
can we use `input` here with type `checkbox` & rename the class to `Checkbox-input` ? will be more semantically correct then.
The `onChange` will be used here then as well.

Reply:
That would require to pile on more hacks than I'm already doing. Hiding the native checkbox while retaining focusability and native behavior of buttons etc. Unless the current solution has some accessibility issues I'd suggest leaving it as is. I'm open someone making the PR if they think it adds any benefit.

Reply:
Sure we can figure it out later, but for now, let's add `role="checkbox` and `aria-checked` so that the checkbox is accessible.

Reply:
done